### PR TITLE
Set up docker-compose

### PR DESCRIPTION
**Pull request by ana0**
Setting up to run Parity and The Graph with docker

### CODE REVIEW

**Review by philipstanislaus — COMMENTED** [review]
Nice, thanks!

Could you add an example of deploying a contract and indexing some data, e. g. the GravatarRegistry (see https://thegraph.com/docs/quick-start#local-development)?

Thanks!

File: e2e-tests/parity/chain.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{
```
[the remark this answers is not in the document]
Comment (ana0) [reply e2e-tests/parity/chain.json:1]: Nope - oops!

File: e2e-tests/parity/config.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[parity]
+# Custom chain
+chain = "/home/parity/.local/share/io.parity.ethereum/chain.json"
```
[the remark this answers is not in the document]
Comment (ana0) [reply e2e-tests/parity/config.toml:3]: This is pointing to a directory inside the official parity docker image, which builds from Ubuntu, not the local machine: https://github.com/openethereum/parity-ethereum/blob/55c90d4016505317034e3e98f699af07f5404b63/scripts/docker/hub/Dockerfile

Could look at changing it but think we can count on the OS here?

[the remark this answers is not in the document]
Comment (philipstanislaus) [reply e2e-tests/parity/config.toml:3]: Okay, that's fine then 👍

File: e2e-tests/Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    $(COMPOSE) up -d
+
+down: ## Stop containers
+    $(COMPOSE) down
+
+logs: ## Follow container logs
+    $(COMPOSE) logs -f --tail 100
+
+clean: ## Remove volumes
+    docker volume prune
```
[the remark this answers is not in the document]
Comment (ana0) [reply e2e-tests/Makefile:18]: Yup, will look into it

File: e2e-tests/Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Prune all volumes and start containers
+start: | clean up
+
+.PHONY: build up down logs clean start
```
[the remark this answers is not in the document]
Comment (ana0) [reply e2e-tests/Makefile:23]: Done!

**Review by philipstanislaus — APPROVED** [review]

File: e2e-tests/docker-compose.yml [hunk 5]
```diff
@@ -49,25 +49,41 @@ services:
     ports:
       - "8020:8020"
     environment:
-      - ethereum=DevelopmentChain:http://parity:8545
+      - ethereum=mainnet:http://parity:8545
```
Comment (philipstanislaus) [inline e2e-tests/docker-compose.yml:52]: Is `mainnet` correct here?

**Review by philipstanislaus — COMMENTED** [review]
Very cool 🎉  👍

File: e2e-tests/docker-compose.yml (hunk 5 above)
Comment (ana0) [reply e2e-tests/docker-compose.yml:52]: Actually, no - but the subgraph.yaml from the example repo I was working with had this and a few other things hardcoded. Eventually I gave up on working around them and forked it! This should probably actually be an env var ... will update

**Review by philipstanislaus — APPROVED** [review]

**Review by philipstanislaus — APPROVED** [review]
LGTM 🙏

File: e2e-tests/contracts/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
+FROM isthisa/tinlake_contracts:v0.1.2
```
Comment (philipstanislaus) [inline e2e-tests/contracts/Dockerfile:1]: How do we run this dockerfile without the dependency on your image?

File: e2e-tests/contracts/deploy.sh [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+PROXY_REGISTRY=$(create_contract tinlake-proxy ProxyRegistry master)
+POOLS_REGISTRY=$(create_contract tinlake-pools-cli PoolRegistry main)
+TINLAKE_CLAIM_RAD=$(create_contract tinlake-claim-rad TinlakeClaimRAD main)
+ACTIONS=$(create_contract tinlake-actions Actions fix/rollback)
+MULTICALL=$(create_contract multicall Multicall master)
+
+# deploy contracts using truffle migrate
+create_contracts_npm centrifuge-ethereum-contracts parity
```
Comment (philipstanislaus) [inline e2e-tests/contracts/deploy.sh:38]: Nicely abstracted 👍

File: e2e-tests/subgraph/pools-metadata.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                "Asset Originator":"Local Revolving Pool LLC",
+                "Collateral":"Simplified invoices",
+                "Average Invoice Size":"$1,000",
+                "Maturity Date":"30 day invoices",
+                "Issuer":"Local Revolving Pool LLC"
+            },
+            "discourseLink":"https://discourse.centrifuge.io/",
+            "asset":"Invoice",
+            "securitizeId":"kickfurther"
```
Comment (philipstanislaus) [inline e2e-tests/subgraph/pools-metadata.js:51]: Can we change this to "localrevolvingpool"? Or is there as specific reason that this is "kickfurther"?

**Review by philipstanislaus — APPROVED** [review]
LGTM 👍

File: e2e-tests/contracts/Dockerfile (hunk 6 above)
Comment (ana0) [reply e2e-tests/contracts/Dockerfile:1]: I think someone on centrifuge side should make a dockerhub account (or maybe you already have one) and push there - the code for this image is tracked here: https://github.com/ana0/tinlake-contracts-docker

File: e2e-tests/subgraph/pools-metadata.js (hunk 8 above)
Comment (ana0) [reply e2e-tests/subgraph/pools-metadata.js:51]: Yes! Will change, I was just copying params from kovan

File: e2e-tests/contracts/deploy.sh (hunk 7 above)
Comment (ana0) [reply e2e-tests/contracts/deploy.sh:38]: Thanks!

File: e2e-tests/contracts/Dockerfile (hunk 6 above)
Comment (philipstanislaus) [reply e2e-tests/contracts/Dockerfile:1]: Nice, could you create a PR into this repo?

**Comment by offerijns** [comment]
Replaced by https://github.com/centrifuge/apps/pull/220.

**Pull request closed without merging**